sale_global_discount/models/sale_order.py

```diff
@@ -0,0 +1,118 @@
+# Copyright 2020 Tecnativa - David Vidal
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+from odoo import api, fields, models
+
+
+class SaleOrder(models.Model):
+    _inherit = 'sale.order'
+
+    global_discount_ids = fields.Many2many(
+        comodel_name='global.discount',
+        string='Sale Global Discounts',
+        domain="[('discount_scope', '=', 'sale'), "
+               "('account_id', '!=', False), '|', "
+               "('company_id', '=', company_id), ('company_id', '=', False)]",
+    )
+    amount_global_discount = fields.Monetary(
+        string='Total Global Discounts',
+        compute='_amount_all',
+        currency_field='currency_id',
+        readonly=True,
+    )
+    amount_untaxed_before_global_discounts = fields.Monetary(
+        string='Amount Untaxed Before Discounts',
+        compute='_amount_all',
+        currency_field='currency_id',
+        readonly=True,
+    )
+    amount_total_before_global_discounts = fields.Monetary(
+        string='Amount Untaxed Before Discounts',
+        compute='_amount_all',
+        currency_field='currency_id',
+        readonly=True,
+    )
+
+    @api.model
+    def get_discounted_global(self, price=0, discounts=None):
```

Comment:
@api.model
    def get_discounted_global(self, price=0, discounts=[]):
        """Compute discounts successively"""
        if not discounts:

Reply:
Sorry, what do you mean? :sweat_smile:

Reply:
He wants you to change the signature function replacing None by [].

But @manuelcalerosolis, this is not good as explained : https://florimond.dev/blog/articles/2018/08/python-mutable-defaults-are-the-source-of-all-evil/